src/components/SelectList/SelectList.tsx

```diff
@@ -21,6 +21,19 @@ import { SelectListProps } from './types';
 
 type WithSelectProps<T> = T & { selectProps: SelectListProps };
 
+const getOptionError = (option: unknown): boolean =>
+    typeof option === 'object' && 'error' in option && Boolean(option.error);
+
+const getOptionVariant = (selectProps: Props, option: unknown): 'default' | 'disabled' | 'error' => {
+    if (selectProps.isDisabled) {
+        return 'disabled';
+    }
+
+    return !getOptionError(option) ? 'default' : 'error';
```

Comment:
We should swap it to keep the "else" branch always showing the default variant. It will be easier to have the default variant chosen in all the non-specific cases